PortalSwift/Sources/PortalSwift/Utils/PortalKeychainAccess.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  PortalKeychainAccess.swift
+//
+//
+//  Created by Blake Williams on 3/31/24.
+//
+
+import Foundation
+
+public class PortalKeychainAccess {
+  private let baseKey = "PortalMpc"
+  private let logger = PortalLogger()
+
+  public func addItem(_ key: String, value: String) throws {
+    // Construct the query to set the keychain item.
+    let query: [String: AnyObject] = [
+      kSecAttrService as String: "\(self.baseKey).\(key)" as AnyObject,
+      kSecAttrAccount as String: key as AnyObject,
+      kSecClass as String: kSecClassGenericPassword,
+      kSecAttrAccessible as String: kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly as AnyObject,
+      kSecValueData as String: value.data(using: String.Encoding.utf8) as AnyObject,
+    ]
+
+    // Try to set the keychain item that matches the query.
+    let status = SecItemAdd(query as CFDictionary, nil)
+
+    // Throw if the status is not successful.
+    if status == errSecDuplicateItem {
+      try self.updateItem(key, value: value)
+    }
+
+    guard status != errSecNotAvailable else {
+      self.logger.error("PortalKeychain.updateItem() - Keychain unavailable: \(status)")
+      throw PortalKeychainAccessError.keychainUnavailableOrNoPasscode(status)
+    }
+    guard status == errSecSuccess else {
+      self.logger.error("PortalKeychain.updateItem() - Unhandled error: \(status)")
+      throw PortalKeychainAccessError.unhandledError(status)
+    }
+  }
+
+  public func deleteItem(_ key: String) throws {
+    let query: [String: AnyObject] = [
+      kSecAttrService as String: "\(self.baseKey).\(key)" as AnyObject,
```

Comment:
Just confirming that we didn't have any backwards compatibility conditions in the previous logic for when we formatted the id differently in the distant past (ie. did we always use this baseKey prefix?)

Reply:
We've always used `PortalMpc` as the `baseKey`, the other backward compatibility checks live in the `PortalKeychain` class itself.